### ISSUE TITLE
Crash on iOS when sending the app to background before AL SDK initialisation

### ISSUE THREAD

**Issue by ltk-daniel-zapater** [AppLovin/AppLovin-MAX-Unity-Plugin#380]
### MAX Plugin Version

6.5.2

### Unity Version

2022.3.22f1

### Device/Platform Info

iOS

### Current Behavior

App crashes when sent to background before AL SDK initialisation

### Expected Behavior

App does not crash

### How to Reproduce

- Send the app to background before AppLovin SDK initialisation
- It also happens if a iOS pop-up alert is shown before AL SDK is initialised

### Additional Info

- The crash happens because backgroundCallback is null in MAUnityAdManager:forwardUnityEventWithArgs
- It also happens in previous AL SDK versions, at least in 6.5.1 and maybe in a few more

**Comment by santoshbagadi** [comment]
@ltk-daniel-zapater thank you for raising the issue. We'll fix this in the next version of our plugin. 

In the meantime, if you'd like to apply the fix, please update the following line in [`Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m`](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/blob/fcffcfb21b5f13b3476d626d34a221ad3c7d499a/DemoApp/Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m#L1996)
```
backgroundCallback(serializedParameters.UTF8String);
```

 to 

```
if ( backgroundCallback )
{
    backgroundCallback(serializedParameters.UTF8String);
}
```

**Comment by ltk-daniel-zapater** [comment]
Thank you for your quick response @santoshbagadi! Our solution was to add this at the beginning of `MAUnityAdManager:forwardUnityEventWithArgs`, so that nothing unnecessary is executed. Wouldn't it be correct?

```
if (backgroundCallback == nil)   
    return;
```

**Comment by santoshbagadi** [comment]
Yes, that would work too. I just wanted to give a quick fix while we are working on a more robust fix for this issue.

**Comment by AndrewUrbanovich** [comment]
Hi @santoshbagadi !
We have exact same issue in Unity 2021.3.37f (although it should be happening in every other unity version too) with AppLovin Max 6.5.2
And this is happening because we don't call MaxSdk.InitializeSdk() if there is no internet connection.
Fix mentioned above with checking if backgroundCallback is configured is working.
But we also have issue with long freezes on iOS when app is going to background and back. (It's not happening on our previous 6.4.4 version). Any advices?

**Comment by AndrewUrbanovich** [comment]
Regarding freezes in our app - it's also happening because of same part of new logic with backgroundCallbackEventsQueue.
If we'll remove part with backgroundCallbackEventsQueue app working much better after returning from background.
 
![image](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/assets/129737735/f76d3d51-3fd9-4ed4-97dc-d2c970f4f9ce)

**Comment by santoshbagadi** [comment]
We've just released a fix for this in [v6.6.0](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/releases/tag/release_6_6_0) of our plugin.
